dockerbuild/Dockerfile-backend

```diff
@@ -0,0 +1,67 @@
+FROM openjdk:8
+
+EXPOSE 6543
+
+# Maven installs openjdk-7 as the default Java. update-alternatives points java back to 8.
+RUN apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv 0C49F3730359A14518585931BC711F9BA15703C6 && \
+    echo "deb [ arch=amd64 ] http://repo.mongodb.org/apt/ubuntu trusty/mongodb-org/3.4 multiverse" | tee /etc/apt/sources.list.d/mongodb-org-3.4.list && \
+    apt-get update && \
+    apt-get install -y \
+        maven \
+        sqlite \
+        g++ \
+        make \
+        automake \
+        bison \
+        flex \
+        pkg-config \
+        libevent-dev \
+        libssl-dev \
+        libtool \
+        mongodb-org-shell && \
+    apt-get clean && \
+    update-alternatives --set java /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java
+
+WORKDIR /root
+
+# Install Thrift 0.10.0
+RUN wget -q http://archive.apache.org/dist/thrift/0.10.0/thrift-0.10.0.tar.gz && \
+    tar -xzf thrift-0.10.0.tar.gz && \
+    cd thrift-0.10.0 && \
+    ./configure --without-python && \
```

Comment:
we only need java for the backend so maybe we can also specify other languages if that helps with faster config too.